src/main/java/org/avarion/graves/listener/PlayerDeathListener.java

```diff
@@ -1,44 +1,48 @@
 package org.avarion.graves.listener;
 
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
 import org.avarion.graves.Graves;
 import org.avarion.graves.manager.CacheManager;
-import org.bukkit.event.EventHandler;
-import org.bukkit.event.EventPriority;
+import org.bukkit.event.Event;
+import org.bukkit.event.EventException;
 import org.bukkit.event.Listener;
 import org.bukkit.event.entity.PlayerDeathEvent;
 import org.bukkit.inventory.ItemStack;
-import org.jetbrains.annotations.NotNull;
+import org.bukkit.plugin.EventExecutor;
 
-import java.util.ArrayList;
-import java.util.Iterator;
-import java.util.List;
-
-public class PlayerDeathListener implements Listener {
+public class PlayerDeathListener implements EventExecutor {
 
     private final Graves plugin;
 
     public PlayerDeathListener(Graves plugin) {
         this.plugin = plugin;
     }
 
-    @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
-    public void onPlayerDeathEvent(@NotNull PlayerDeathEvent event) {
-        List<ItemStack> itemStackList = event.getDrops();
+    @Override
+    public void execute(Listener listener, Event event) throws EventException {
```

Comment:
Oh, we don't have an `onPlayerDeathEvent` anymore if we dynamically add that priority?

Reply:
Yes, to make priority configurable you have to use the old way of registering events using `PluginManager#registerEvent` instead of `PluginManager#registerEvents` - see https://hub.spigotmc.org/stash/projects/SPIGOT/repos/bukkit/browse/src/main/java/org/bukkit/plugin/java/JavaPluginLoader.java#302 for Spigot's own code

Reply:
Pitty, but thanks for clarifying that!